### PR TITLE
Add grpc back

### PR DESCRIPTION
**Pull request by RmStorm**

### CODE REVIEW

**Comment by RmStorm** [comment]
I'm not sure if I made this one: https://github.com/kolonialno/aviso/pull/696 dependent on this PR :sweat_smile: some last stuff to straighten out I think.

**Comment by gbbirkisson** [comment]
I see this is still a draft, but I have a suggestion. Let the user of troncos decide to install either `opentelemetry-exporter-otlp-proto-grpc` or `opentelemetry-exporter-otlp-proto-http`. Then make troncos pick based on what is installed.

File: troncos/traces/__init__.py [hunk 1]
```diff
@@ -1,26 +1,47 @@
...
 _GLOBAL_SPAN_PROCESSORS_SET_ONCE = Once()
 
-_DEBUG_SPAN_PROCESSOR: SpanProcessor = SimpleSpanProcessor(ConsoleSpanExporter())
+if out_file := os.environ.get("TRACING_DEBUGGING_FILE", None):
+    traces_out: IO[str] = open("traces.json", "w")
+else:
+    traces_out = sys.stdout
+_DEBUG_SPAN_PROCESSOR: SpanProcessor = SimpleSpanProcessor(
+    ConsoleSpanExporter(out=traces_out)
+)
```
Comment (RmStorm) [inline troncos/traces/__init__.py:39]: trace volume can be pretty intense. Much nicer to shove it in a file than in the terminal.

File: troncos/traces/dd_exporter.py [hunk 2]
```diff
@@ -1,6 +1,16 @@
 from typing import Sequence
 
+try:
+    from opentelemetry.exporter.otlp.proto.grpc import (  # type: ignore
+        trace_exporter as grpc_trace_exporter,
+    )
+except ImportError:  # pragma: no cover
+    grpc_trace_exporter = None
```
Comment (RmStorm) [inline troncos/traces/dd_exporter.py:8]: This is how it's done in Aviso as well: https://github.com/kolonialno/aviso/blob/main/aviso-python/src/aviso/tracing/_tracer.py

File: troncos/traces/dd_exporter.py [hunk 3]
```diff
@@ -9,17 +19,30 @@
...
+    )
+    sdk_span._instrumentation_info = InstrumentationInfo(
+        name=sdk_span.name,
+        version="0.0.0",
+    )
+    if "resource" in sdk_span.attributes:  # type: ignore[operator]
+        sdk_span._name = sdk_span.attributes["resource"]  # type: ignore # noqa: E501
+
+
+class OTLPGrpcSpanExporterDD(grpc_trace_exporter.OTLPSpanExporter if grpc_trace_exporter else object):  # type: ignore  # noqa: E501
```
Comment (RmStorm) [inline troncos/traces/dd_exporter.py:35]: yeah, idk.. awesome parent class :see_no_evil: it does work though!

File: troncos/traces/__init__.py [hunk 4]
```diff
@@ -100,6 +132,9 @@ def init_tracing_provider(
     if global_provider:
         opentelemetry.trace.set_tracer_provider(provider)
 
+    if os.environ.get("TRACING_DEBUGGING_FILE"):
+        init_tracing_debug(provider)
```
Comment (gbbirkisson) [inline troncos/traces/__init__.py:136]:
```suggestion
```

It's strange to add debugging here only if you specify `TRACING_DEBUGGING_FILE` but if you don't, you have to call `init_tracing_debug` yourself. I vote removing this.

File: troncos/traces/dd_exporter.py (hunk 3 above)
Comment (gbbirkisson) [reply troncos/traces/dd_exporter.py:35]: Not the prettiest thing no, but these are internals that we can change later if we want,

**Review by gbbirkisson — APPROVED** [review]

**Pull request merged**